components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/internal/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -612,7 +612,7 @@ private void addActionPropertiesToDB(String actionId, Map<String, ActionProperty
                                 statement.setNull(ActionMgtSQLConstants.Column.ACTION_PROPERTIES_PRIMITIVE_VALUE,
                                         java.sql.Types.VARCHAR);
                                 statement.setBinaryStream(ActionMgtSQLConstants.Column.ACTION_PROPERTIES_OBJECT_VALUE,
-                                        ((BinaryObject) property.getValue().getValue()).getInputStream(),
+                                        ((BinaryObject) property.getValue().getValue()).getStreamValue(),
```

Comment:
How do you decide whether to call "getStreamValue" or "getStringValue" ?. How do you know, from which object type the object has been creared ?